bool pc
int8 command_source
int8 command_type
string command